### DISCUSSION TITLE
webhook - infinite loop

### DISCUSSION THREAD

**Discussion by jnc27** [netbox-community/netbox#13136] [Q&A] [answered]
I am using webhook to update custom data fields whenever a user changes some data in a device. Webhook is installed directly on the netbox server.
Updating netbox custom data field seems to work, however I am running into an infinite loop, where webhook detects the http post request constantly and thus produce an infinite update of the device.

What is the best way to avoid infinite loops? 

Thank you.

**Comment by markkuleinio** [comment]
One solution is to use a specific username (its API token) in the webhook handler, and then compare the username field in the incoming webhook and just ignore the call if the username matches the webhook handler username.

**Reply by jnc27** [reply]
Thanks for the answer.
I think I understand the approach. But how would I retrieve the username?

**Reply by markkuleinio** [reply, marked as answer]
In the default webhook request body there is the `username` field: https://docs.netbox.dev/en/stable/integrations/webhooks/#default-request-body

**Reply by jnc27** [reply]
got you! Thanks a lot!
